### PR TITLE
Fix a few typos

### PR DESCRIPTION
**Pull request by kinow**
*Issue #, if available:*

None.

*Description of changes:*

Fixed typos mainly in docstrings, but also in some messages printed to users I think.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Yup!

Thanks!

### CODE REVIEW

File: awscli/customizations/s3/filters.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ class Filter(object):
     """
     def __init__(self, patterns, rootdir, dst_rootdir):
         """
-        :var patterns: A list of patterns. A pattern consits of a list
+        :var patterns: A list of patterns. A pattern consist of a list
```
Comment (JordonPhillips) [inline awscli/customizations/s3/filters.py]: This should be plural

File: awscli/customizations/s3/results.py [hunk 2]
```diff
@@ -286,7 +286,7 @@ def _record_progress_result(self, result, **kwargs):
         # gets created that the timestamp on the progress result is less
         # than the timestamp of when the result processor actually
         # processes that initial queued result. So this will avoid
-        # negative progress being displayed or zero divison occuring.
+        # negative progress being displayed or zero divisionoccurringg.
```
Comment (JordonPhillips) [inline awscli/customizations/s3/results.py]: This one doesn't seem right

File: awscli/customizations/s3/filters.py (hunk 1 above)
Comment (kinow) [reply awscli/customizations/s3/filters.py]: Fixed!

File: awscli/customizations/s3/results.py (hunk 2 above)
Comment (kinow) [reply awscli/customizations/s3/results.py]: Oh, sorry, fixed too! Thanks!

**Comment by kinow** [comment]
Typos in the pull request to fix typos fixed :slightly_smiling_face:  Thanks @JordonPhillips . Edited commit, rebased, push-force'd.

**Comment by kinow** [comment]
> Please fix merge conflict.

Rebased, fixed conflicts, push-forced to my branch :+1:

**Review by justindho — APPROVED** [review]
@kinow Thanks for the PR!

**Pull request merged**